types/mithril/index.d.ts

```diff
@@ -1,863 +1,268 @@
-		* case the response contains metadata wrapping the data.
-		*
-		* @param data The data to unwrap.
-		* @return The unwrapped result.
-		*/
-		unwrapSuccess?(data: any): any;
-
-		/**
-		* An optional preprocessor function to unwrap an unsuccessful response,
-		* in case the response contains metadata wrapping the data.
-		*
-		* @param data The data to unwrap.
-		* @return The unwrapped result.
-		*/
-		unwrapError?(data: any): any;
-
-		/**
-		* An optional function to serialize the data. This defaults to
-		* `JSON.stringify`.
-		*
-		* @param dataToSerialize The data to serialize.
-		* @return The serialized form as a string.
-		*/
-		serialize?(dataToSerialize: any): string;
-
-		/**
-		* An optional function to deserialize the data. This defaults to
-		* `JSON.parse`.
-		*
-		* @param dataToSerialize The data to parse.
-		* @return The parsed form.
-		*/
-		deserialize?(dataToDeserialize: string): any;
-
-		/**
-		* An optional function to extract the data from a raw XMLHttpRequest,
-		* useful if the relevant data is in a response header or the status
-		* field.
-		*
-		* @param xhr The associated XMLHttpRequest.
-		* @param options The options passed to this request.
-		* @return string The serialized format.
-		*/
-		extract?(xhr: XMLHttpRequest, options: this): string;
-
-		/**
-		* The parsed data, or its children if it's an array, will be passed to
-		* this class constructor if it's given, to parse it into classes.
-		*
-		* @param data The data to parse.
-		* @return The new instance for the list.
-		*/
-		type?: new (data: any) => any;
-
-		/**
-		* The URL to send the request to.
-		*/
-		url: string;
+	/** Components are a mechanism to encapsulate parts of a view to make code easier to organize and/or reuse. Any Javascript object that has a view method is a Mithril component. Components can be consumed via the m() utility. */
+	export interface Component<Attrs, State extends Lifecycle<Attrs, State>> extends Lifecycle<Attrs, State> {
+		/** Creates a view out of virtual elements. */
+		view(this: State, vnode: Vnode<Attrs, State>): Children | null | void;
 	}
 
-	/**
-	* This represents the available options for configuring m.request for JSONP
-	* requests.
-	*
-	* @see m.request
-	*/
-	interface JSONPOptions extends RequestOptions {
-		/**
-		* For JSONP requests, this must be the string "jsonp". Otherwise, it's
-		* ignored.
-		*/
-		dataType: "jsonp";
-
-		/**
-		* The querystring key for the JSONP request callback. This is useful for
-		* APIs that don't use common conventions, such as
-		* `www.example.com/?jsonpCallback=doSomething`. It defaults to
-		* `callback`.
-		*/
-		callbackKey?: string;
-
-		/**
-		 * The data to send with the request. This is automatically serialized
-		 * to a querystring.
-		 */
-		data?: Object;
+	export interface ClassComponent<A> extends Lifecycle<A, ClassComponent<A>> {
+		view(this: ClassComponent<A>, vnode: CVnode<A>): Children | null | void;
```

Comment:
Drop the `this` parameter: it defaults to the correct `this` type of whatever the real resulting type is.

Oh, and just duplicate the `Lifecycle` attributes here, to work around the fact you currently can't specify a generic as `this`, so the type is actually correct.